serde_someip/src/ser.rs

```diff
@@ -395,110 +382,212 @@ impl<'a> ser::SerializeStructVariant for &'a mut Serializer {
     }
 
     #[test]
-    fn test_char(){
+    fn test_char() {
         let char: char = 'V';
         let mut ex = vec![];
         ex.push(char as u8);
-        assert_eq!(to_bytes(&char).unwrap(),ex);
+        assert_eq!(to_bytes(&char).unwrap(), ex);
     }
 
     #[test]
-    fn test_seq(){
-        let test:Vec<u8> = vec![1,2];
-        let expected =vec![1,2];
+    fn test_seq() {
+        let test: Vec<u8> = vec![1, 2];
+        let expected = vec![1, 2];
         assert_eq!(to_bytes(&test).unwrap(), expected);
     }
 
     #[test]
-    fn test_tuple(){
-        let test:(u8,u8,u8) = (1,2,3);
-        let expected:Vec<u8> = vec![1,2,3];
+    fn test_nested_seq() {
+        let test: Vec<Vec<u8>> = vec![vec![1, 3], vec![3, 4], vec![5, 6]];
+        let expected: Vec<u8> = vec![1, 3, 3, 4, 5, 6];
         assert_eq!(to_bytes(&test).unwrap(), expected);
     }
 
+    #[test]
+    fn test_tuple() {
+        let test: (u8, u8, u8) = (1, 2, 3);
+        let expected: Vec<u8> = vec![1, 2, 3];
+        assert_eq!(to_bytes(&test).unwrap(), expected);
+    }
+
+    #[test]
+    fn test_nested_tuple() {
+        let test: ((u8, u8), (u8, u8)) = ((1, 2), (2, 3));
+        let expected: Vec<u8> = vec![1, 2, 2, 3];
+        assert_eq!(to_bytes(&test).unwrap(), expected);
+    }
 
     #[test]
     fn test_struct() {
         #[derive(Serialize)]
         struct Test {
             int: u8,
-            bool:bool,
-            str:String,
+            bool: bool,
+            float: f32,
+            char: char,
+            str: String,
+            tup: (u8, u8),
             seq: Vec<&'static str>,
+            seq2: Vec<u8>,
+            seq3: Vec<Vec<u8>>,
         }
 
         let test = Test {
             int: 98,
             bool: false,
-            str:String::from("test"),
+            float: 342.1274,
+            char: 'c',
+            str: String::from("test"),
+            tup: (1, 3),
             seq: vec!["a", "b"],
+            seq2: vec![12, 32],
+            seq3: vec![vec![12, 34], vec![23, 43]],
         };
-        
-        let expected = vec![98,0,116, 101, 115, 116,97,98];
+
+        let expected = vec![
+            98, 0, 79, 16, 171, 67, 99, 7, 0, 0, 0, 239, 187, 191, 116, 101, 115, 116, 0, 1, 3, 4,
+            0, 0, 0, 239, 187, 191, 97, 0, 4, 0, 0, 0, 239, 187, 191, 98, 0, 12, 32, 12, 34, 23,
+            43,
+        ];
         assert_eq!(to_bytes(&test).unwrap(), expected);
     }
+
+    #[test]
+    fn test_struct_array() {
+        #[derive(Serialize)]
+        struct Test {
+            seq: [u8; 4],
+        }
+
+        let test = Test {
+            seq: [12, 2, 43, 9],
+        };
+
+        let expected = vec![12, 2, 43, 9];
+        assert_eq!(to_bytes(&test).unwrap(), expected);
+    }
+
+    //negative test cases
+    #[test]
+    fn test_string_null_termination() {
+        let str: String = String::from("Hi");
+        let mut buffer: Vec<u8> = Vec::new();
```

Comment:
Please write a comment describing what this test is doing.  
Instead of assert_ne!,  check for a specific error.